sway-lib-std/src/option.sw

```diff
@@ -1,18 +1,86 @@
-//! Error handling with the `Option` type.
+//! Optional values.
 //!
-//! [`Option<T>`][`Option`] is the type used for representing the existence or absence of a value. It is an enum with the variants, [`Some(T)`], representing
-//! some value, and [`None()`], representing
-//! no value.
+//! Type [`Option`] represents an optional value: every [`Option`]
+//! is either [`Some`] and contains a value, or [`None`], and
+//! does not. [`Option`] types are very common in Sway code, as
+//! they have a number of uses:
+//!
+//! * Initial values
```

Comment:
Maybe expand this to "Initial values can be set to [`None`]"  or something like that.